app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt

```diff
@@ -783,3 +821,35 @@ class TabTrayItemMenu(
         )
     }
 }
+
+class MultiselectSelectionMenu(
+    private val context: Context,
+    private val onItemTapped: (MultiselectSelectionMenu.Item) -> Unit = {}
+) {
+    sealed class Item {
+        object BookmarkTabs : Item()
+        object DeleteTabs : Item()
+    }
+
+    val menuBuilder by lazy { BrowserMenuBuilder(menuItems) }
+
+    private val menuItems by lazy {
+        listOf(
+            SimpleBrowserMenuItem(
+                context.getString(R.string.tab_tray_multiselect_menu_item_bookmark),
+                textColorResource = R.color.primary_text_normal_theme
+            ) {
+                context.components.analytics.metrics.track(Event.TabsTraySaveToCollectionPressed)
+                onItemTapped.invoke(Item.BookmarkTabs)
+            },
+
+            SimpleBrowserMenuItem(
+                context.getString(R.string.tab_tray_multiselect_menu_item_close),
+                textColorResource = R.color.primary_text_normal_theme
+            ) {
+                context.components.analytics.metrics.track(Event.TabsTrayShareAllTabsPressed)
+                onItemTapped.invoke(Item.DeleteTabs)
+            }
+        )
+    }
+}
```

Comment:
Can we move this to a new file `MultiselectSelectionMenu.kt` to reduce the size of the existing file?